packages/diagrams/frontend/sirius-components-diagrams/src/renderer/direct-edit/useDiagramDirectEdit.tsx

```diff
@@ -29,20 +29,30 @@ export const useDiagramDirectEdit = (): UseDiagramDirectEditValue => {
 
   const onDirectEdit = useCallback(
     (event: React.KeyboardEvent<Element>) => {
-      const { key } = event;
+      const { key, target } = event;
       /*If a modifier key is hit alone, do nothing*/
-      const isTextField = event.target instanceof HTMLInputElement || event.target instanceof HTMLTextAreaElement;
-      if ((event.altKey && key === 'Alt') || (event.shiftKey && key === 'Shift') || isTextField || readOnly) {
+      if ((event.altKey && key === 'Alt') || (event.shiftKey && key === 'Shift') || readOnly) {
+        return;
+      }
+
+      /*When the palette is opened, allow direct edit only with F2 so we can type in the palette search field*/
```

Comment:
Why break the existing behavior that has worked for years on both desktop and web?

Reply:
Even that does not work properly, I can successfully bypass it sometimes to restore the original behavior.

Reply:
I'm not sure it's comparable with the desktop version. Here when we open the palette we want to have the focus on the searchfield input and as such we can use direct edit only with F2 if the palette is opened.
If the palette is closed then F2 or typing can be used to start the direct edit.

Do you mean the direct-edit is starting when typing on the searchfield with the PR?